src/components/algorithm/AddAlgorithm.js

```diff
@@ -0,0 +1,234 @@
+  };
+
+  handleBrowseFileCallback = (filePaths) => {
+    // currently we select only one file
+    const filePath = filePaths[0];
+    if (filePath) {
+      this.setState({ fileLocation: filePath });
+    }
+  };
+
+  handleCodeOnChange = (code) => {
+    this.setState({ code });
+  };
+
+  handleOptionOnChange = (event) => {
+    this.setState({ option: event.target.value });
+  };
+
+  handleBack = () => {
+    const {
+      history: { goBack },
+    } = this.props;
+    goBack();
+  };
+
+  handleSave = () => {
+    const { dispatchAddAlgorithm } = this.props;
+    const { name, description, fileLocation, code, option } = this.state;
+    const author = AUTHOR_USER;
+    const payload =
+      option === OPTION_FILE
+        ? { name, description, author, fileLocation }
+        : { name, description, author, code };
+    if (name) {
+      dispatchAddAlgorithm(payload);
+    }
+  };
+
+  render() {
+    const { t, classes } = this.props;
+    const { name, description, option, fileLocation, code } = this.state;
+
+    return (
+      <Main>
+        <Container>
+          <h1>{t('Add algorithm')}</h1>
+          <Grid container spacing={5}>
+            <Grid item xs={option === OPTION_FILE ? 5 : 7}>
+              <FormControl component="fieldset">
+                <RadioGroup
+                  aria-label="add option"
+                  name="add option"
+                  value={option}
+                  onChange={this.handleOptionOnChange}
+                >
+                  <FormControlLabel
+                    value={OPTION_FILE}
+                    control={<Radio color="primary" />}
+                    label={t('Load algorithm from a file')}
+                  />
+                  <FormControlLabel
+                    value={OPTION_EDITOR}
+                    control={<Radio color="primary" />}
+                    label={t('Write algorithm')}
+                  />
+                </RadioGroup>
+              </FormControl>
+              {option === OPTION_FILE ? (
+                <Grid container alignItems="center">
+                  <Grid item xs={11}>
+                    <TextField
+                      margin="dense"
+                      label={t('Select algorithm')}
+                      onChange={this.handleLocationInput}
+                      value={fileLocation}
+                      helperText={t('(Required)')}
+                      required
+                      fullWidth
+                    />
+                  </Grid>
+                  <Grid item xs={1}>
+                    <BrowseFileButton
+                      options={{
+                        filters: [{ name: 'python', extensions: ['py'] }],
+                      }}
+                      onBrowseFileCallback={this.handleBrowseFileCallback}
+                    />
+                  </Grid>
+                </Grid>
+              ) : (
+                <Editor
+                  programmingLanguage="python"
```

Comment:
Refactor `python`. It would be maybe nice to have `programmingLanguage` in the state to later use it to toggle language when we will support more of them.